spec/mempool/v0/quint/mempoolv0.qnt

```diff
@@ -0,0 +1,524 @@
+            // We don't want to repeat this step on the same transaction as the last time.
+            require(_step.name == "ReceiveTxViaRPC" and _step.node == node implies _step.args.get("tx") != tx),
+            node.CheckTxRPC(tx),
+        }
+
+    /* Receive some transaction from a client via RPC: only a transaction that
+     * was not sent before. */
+    action ReceiveNonRepeatedTxViaRPC(node) =
+        val availableTxs = Txs.exclude(History::txsReceivedViaRPC.mapValues().flatten())
+        all {
+            require(availableTxs.nonEmpty()),
+            nondet tx = oneOf(Txs.exclude(History::txsReceivedViaRPC.mapValues().flatten()))
+            node.CheckTxRPC(tx)
+        }
+
+    //--------------------------------------------------------------------------
+    // Consensus actions
+    //--------------------------------------------------------------------------
+    // Consensus reactor's BlockExecutor updates the mempool with the latest
+    // block it created and rechecks outstanding txs.
+    
+    // This node is not the proposer: take the block from the chain.
+    action BlockExecutor_MempoolUpdate_nonProposer(node) = all {
+        require(not(Chain::chainIsEmpty)),
+        require(node.MempoolHeight() < Chain::latestHeight),
+        node.setStep("UpdateAsNonProposer", Map("node" -> node)),
+        val nextBlockHeight = node.MempoolHeight() + 1
+        val txs: Set[Tx] = Chain::getBlockAt(nextBlockHeight)
+        // All txs are valid because they are committed in the chain.
+        node.Update(nextBlockHeight, txs, txs.mapBy(_ => true)),
+        Chain::unchanged,
+    }
+
+    //-- __subaction__: this action is not part of the next-state relation
+    action BlockExecutor_MempoolUpdate_proposer_(node, txs, nextBlockHeight) = all {
+        node.setStep("UpdateAsProposer", Map("node" -> node)),
+        val txValidResults: Tx -> bool = txs.mapBy(tx => tx.isValidAt(nextBlockHeight))
+        node.Update(nextBlockHeight, txs, txValidResults),
+        Chain::addNewBlock(txs),
+    }
+
+    // This node is the proposer: create the next block with txs from the mempool.
+    action BlockExecutor_MempoolUpdate_proposer(node) = all {
+        require(node.mempoolTxs().nonEmpty()),
+        require(node.MempoolHeight() == Chain::latestHeight),
+        val nextBlockHeight = node.MempoolHeight() + 1
+        // Note that block may be empty.
+        nondet txs = oneOf(node.mempoolTxs().powerset())
+        BlockExecutor_MempoolUpdate_proposer_(node, txs, nextBlockHeight)
+    }
+
+    //--------------------------------------------------------------------------
+    // Mempool reactor (P2P) actions
+    //--------------------------------------------------------------------------
+    /* Receive a Txs message from a peer, and calls CheckTx on each Tx. */
+    // [Reactor.Receive]: https://github.com/CometBFT/cometbft/blob/111d252d75a4839341ff461d4e0cf152ca2cc13d/mempool/reactor.go#L93
+    action P2P_ReceiveTxs(node) = all {
```

Comment:
That's correct. I will rename it. As a side note, the protobuf message is called `Txs` and it can contain an array of transactions, but the implementation of v0 only sends one transaction each time. At some moment it was possible to send batches but this was discarded for some reason.